feat(wg-easy): Add WG-Easy Chart

StevenMcElligott opened the pull request:
Initial run of creating my own chart from "scratch". Let's see if it runs, especially with the sysctl

**Description**

Adds WG-Easy to the incubator train

⚒️ Fixes  #2950  

**⚙️ Type of change**

- [X] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [X] ⚖️ My code follows the style guidelines of this project
- [X] 👀 I have performed a self-review of my own code
- [X] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [X] ⬆️ I increased versions for any altered app according to semantic versioning

StevenMcElligott committed:
Add WG-Easy Chart

Initial run of creating my own chart from "scratch". Let's see if it runs, especially with the sysctl

stavros-k commented on charts/incubator/wg-easy/Chart.yaml:
```diff
@@ -0,0 +1,30 @@
+deprecated: false
+description: The easiest way to run WireGuard VPN + Web-based Admin UI.
+home: https://github.com/truecharts/apps/tree/master/charts/stable/jackett
+icon: https://truecharts.org/_static/img/appicons/jackett.png
```
```suggestion
home: https://github.com/truecharts/apps/tree/master/charts/stable/wg-easy
icon: https://truecharts.org/_static/img/appicons/wg-easy.png
```

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,41 @@
+  runAsNonRoot: false
+    sysctls:
+        - name: net.ipv4.conf.all.src_valid_mark
+            value: '1'
```
```suggestion
  runAsNonRoot: false
  sysctls:
    - name: net.ipv4.conf.all.src_valid_mark
      value: '1'
```

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,41 @@
+
+service:
+  main:
+    protocol: TCP
```
```suggestion
```
You can omit this, this is the default ;)

StevenMcElligott replied:
This I didn't know lol

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,41 @@
+persistence:
+  config:
+    enabled: true
+    mountPath: "/etc/wireguard"
```
```suggestion
    mountPath: "/etc/wireguard"

```

StevenMcElligott committed:
Apply suggestions from code review

Thanks for fixing the icons lol

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

StevenMcElligott reviewed (commented):
Added the fixes as suggested by @stavros-k , thanks again!

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,40 @@
+        vpn:
+        protocol: UDP
+        port: 51820
+        targetPort: 51820
```
```suggestion
  vpn:
    enabled: true
    ports:
      vpn:
        protocol: UDP
        port: 51820
        targetPort: 51820
```

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,40 @@
+        targetPort: 51820
+        
+      
+
```
```suggestion
```

stavros-k committed:
Apply suggestions from code review

stavros-k committed:
Update charts/incubator/wg-easy/Chart.yaml

stavros-k committed:
its always that enable :D

StevenMcElligott committed:
Update values.yaml

Hopefully fixing the quotation marks fixes the sysctls error, otherwise I'm putting this in the wrong place lol

StevenMcElligott committed:
Merge branch 'truecharts:master' into master

StevenMcElligott committed:
Remove the likely unnecessary sysctl

Let’s see how it runs without net.ipv4.conf.all.src_valid_mark since you can’t set it kubernetes

StevenMcElligott committed:
Update values.yaml

StevenMcElligott committed:
Merge branch 'master' of https://github.com/StevenMcElligott/TrueChartsApps

StevenMcElligott committed:
Fixed the repository for wg-easy

Had wrong URL and their latest version was changed 4 hours ago

StevenMcElligott committed:
Added missing env-vars to wg-easy

Needs WG_HOST and PASSWORD, the rest people can add themselves

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,42 @@
+  main:
+    ports:
+      main:
+        targetPort: 51821
```
```suggestion
secretEnv:
  PASSWORD: "secretpass"

env:
  WG_HOST: "hostname"

service:
  main:
    ports:
      main:
        targetPort: 51821
```

stavros-k committed:
Update charts/incubator/wg-easy/values.yaml

stavros-k committed:
recommit

stavros-k committed:
emtpy space

StevenMcElligott committed:
Merge branch 'truecharts:master' into master

StevenMcElligott committed:
Added PUID and PGID to values

Added more stuff to see if the container will pass tests

StevenMcElligott committed:
Merge branch 'master' of https://github.com/StevenMcElligott/TrueChartsApps

StevenMcElligott committed:
Merge branch 'truecharts:master' into master

StevenMcElligott committed:
Fixed trailing spaces

StevenMcElligott commented:
If anyone has the time I'm assuming the pull_request is an error in my questions.yaml but not sure what yet lol. At least it ran further than last time haha

stavros-k committed:
identation

stavros-k committed:
Merge branch 'master' into master

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,47 @@
+podSecurityContext:
+  runAsUser: 0
+  runAsGroup: 0
+  fsGroup: 568
```
```suggestion
```

stavros-k replied:
fsGroup by default is 568

stavros-k commented on charts/incubator/wg-easy/values.yaml:
```diff
@@ -0,0 +1,47 @@
+
+security:
+  PUID: 0
+  PGID: 0
```
```suggestion
```

PGID = fsGroup

StevenMcElligott committed:
Update charts/incubator/wg-easy/values.yaml

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

StevenMcElligott committed:
Update charts/incubator/wg-easy/values.yaml

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented on charts/incubator/wg-easy/questions.yaml:
```diff
@@ -0,0 +1,362 @@
+      type: dict
+      attrs:
+        - variable: hostname
+          label: "hostname"
```
```suggestion
        - variable: WG_HOST
          label: "WG_HOST"
```

stavros-k committed:
Update charts/incubator/wg-easy/questions.yaml

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
